src/bonefish/messages/wamp_message.hpp

```diff
@@ -28,22 +29,47 @@ class wamp_message
 {
 public:
     wamp_message();
-    virtual ~wamp_message();
+    virtual ~wamp_message() = default;
 
-    wamp_message(const wamp_message& other) = delete;
+    wamp_message(const wamp_message&) = delete;
+    wamp_message(wamp_message&&) = delete;
     wamp_message& operator=(wamp_message const&) = delete;
+    wamp_message& operator=(wamp_message&&) = delete;
+
+    std::unique_ptr<msgpack::zone>&& release_zone();
```

Comment:
That by itself is not really much of a hindrance, as you could just do `std::move(*unpacker.zone())` instead of moving its pointer :P but I'll leave it up to you which one to prefer in the end.